aiohttp/web_exceptions.py

```diff
@@ -83,23 +88,71 @@ class HTTPException(Response, Exception):
     def __init__(self, *,
                  headers: Optional[LooseHeaders]=None,
                  reason: Optional[str]=None,
-                 body: Any=None,
                  text: Optional[str]=None,
                  content_type: Optional[str]=None) -> None:
-        if body is not None:
-            warnings.warn(
-                "body argument is deprecated for http web exceptions",
-                DeprecationWarning)
-        Response.__init__(self, status=self.status_code,
-                          headers=headers, reason=reason,
-                          body=body, text=text, content_type=content_type)
-        Exception.__init__(self, self.reason)
-        if self.body is None and not self.empty_body:
-            self.text = "{}: {}".format(self.status, self.reason)
+        if reason is None:
+            try:
+                reason = RESPONSES[self.status_code][0]
```

Comment:
Maybe add method for this with lru cache?

Reply:
Isn't it faster in case of exception handling branch invocation?

Reply:
Are you talking about `lru_cache` decorator from `functools`? This decorator is not for free.
Anyway, I'll switch to direct `HTTPStatus` usage.